src/VecSim/algorithms/hnsw/hnsw.h

```diff
@@ -96,199 +93,203 @@ class HierarchicalNSW : public VecsimBaseObject {
+    candidatesLabelsMaxHeap<DistType>
+    searchBottomLayer_WithTimeout(idType ep_id, const void *data_point, size_t ef, size_t k,
+                                  void *timeoutCtx, VecSimQueryResult_Code *rc) const;
     VecSimQueryResult *searchRangeBottomLayer_WithTimeout(idType ep_id, const void *data_point,
-                                                          double epsilon, dist_t radius,
+                                                          double epsilon, DistType radius,
                                                           void *timeoutCtx,
                                                           VecSimQueryResult_Code *rc) const;
-    void getNeighborsByHeuristic2(candidatesMaxHeap<dist_t> &top_candidates, size_t M);
-    idType mutuallyConnectNewElement(idType cur_c, candidatesMaxHeap<dist_t> &top_candidates,
-                                     size_t level);
+    void getNeighborsByHeuristic2(candidatesMaxHeap<DistType> &top_candidates, size_t M);
+    inline idType mutuallyConnectNewElement(idType cur_c,
+                                            candidatesMaxHeap<DistType> &top_candidates,
+                                            size_t level);
     void repairConnectionsForDeletion(idType element_internal_id, idType neighbour_id,
                                       idType *neighbours_list, idType *neighbour_neighbours_list,
                                       size_t level, vecsim_stl::vector<bool> &neighbours_bitmap);
-    void replaceEntryPoint(idType element_internal_id);
-    void SwapLastIdWithDeletedId(idType element_internal_id, idType last_element_internal_id);
+    inline void replaceEntryPoint();
+    inline void SwapLastIdWithDeletedId(idType element_internal_id,
+                                        idType last_element_internal_id);
 
 public:
-    HierarchicalNSW(const HNSWParams *params, dist_func_t<dist_t> func,
-                    std::shared_ptr<VecSimAllocator> allocator, size_t random_seed = 100,
-                    size_t initial_pool_size = 1);
-    virtual ~HierarchicalNSW();
+    HNSWIndex(const HNSWParams *params, std::shared_ptr<VecSimAllocator> allocator,
+              size_t random_seed = 100, size_t initial_pool_size = 1);
+    virtual ~HNSWIndex();
 
-    void setEf(size_t ef);
-    size_t getEf() const;
+    inline void setEf(size_t ef);
+    inline size_t getEf() const;
     inline void setEpsilon(double epsilon);
     inline double getEpsilon() const;
-    size_t getIndexSize() const;
-    size_t getIndexLabelCount() const;
-    size_t getIndexCapacity() const;
-    size_t getEfConstruction() const;
-    size_t getM() const;
-    size_t getMaxLevel() const;
-    size_t getEntryPointLabel() const;
-    idType getEntryPointId() const;
-    labelType getExternalLabel(idType internal_id) const;
-    VisitedNodesHandler *getVisitedList() const;
+    inline size_t indexSize() const override;
+    inline size_t indexLabelCount() const override;
```

Comment:
why the renaming?
getX
setX

Reply:
@GuyAv46 ?

Reply:
To align with BF and the `VecSimIndexAbstract` requirements. it can also be the other way around.

Reply:
It was already `indexSize()` and `indexLabelCount()` in the wrapper